doc/src/manual/environment-variables.md

```diff
@@ -0,0 +1,312 @@
+
+For example, a Linux installation with a Julia executable located at
+`/bin/julia`, a `DATAROOTDIR` of `../share`, and a `SYSCONFDIR` of `../etc`
+will have `JULIA_HOME` set to `/bin`, a source-file search path of
+
+```
+/share/julia/base
+```
+
+and a global configuration search path of
+
+```
+/etc/julia/juliarc.jl
+```
+
+### `JULIA_LOAD_PATH`
+
+A separated list of absolute paths that are to be appended to the variable
+[`Base.LOAD_PATH`](@ref). (In Unix-like systems, the path separator is `:`; in
+Windows systems, the path separator is `;`.) The `Base.LOAD_PATH` variable is
+where [`Base.require`](@ref) and `Base.load_in_path()` look for
+code; it defaults to the absolute paths
+
+```
+$JULIA_HOME/../local/share/julia/site/v$(Base.VERSION.major).$(Base.VERSION.minor)
+$JULIA_HOME/../share/julia/site/v$(Base.VERSION.major).$(Base.VERSION.minor)
+```
+
+so that, e. g., version 0.5 of Julia on a Linux system with a Julia executable
+at `/bin/julia` will have a default `Base.LOAD_PATH` of
+
+```
+/local/share/julia/site/v0.5
+/share/julia/site/v0.5
+```
+
+### `JULIA_PKGDIR`
+
+The path of the parent directory `Dir._pkgroot()` for the version-specific
+Julia package repositories. If the path is relative, then it is taken with
+respect to the working directory. If `$JULIA_PKGDIR` is not set, then
+`_pkgroot()` defaults to
+
+```
+$HOME/.julia
+```
+
+Then the repository location [`Base.Pkg.dir`](@ref) for a given
+Julia version is
+
+```
+$JULIA_PKGDIR/v$(Base.VERSION.major).$(Base.VERSION.minor)
+```
+
+For example, for a Linux user whose home directory is `/home/alice`, the
+directory containing the package repositories would by default be
+
+```
+/home/alice/.julia
+```
+
+and the package repository for version 0.5 of Julia would be
+
+```
+/home/alice/.julia/v0.5
+```
+
+### `JULIA_HISTORY`
+
+The absolute path `Base.find_hist_file()` of the REPL's history file. If
+`$JULIA_HISTORY` is not set, then `Base.find_hist_file()` defaults to
+
+```
+$HOME/.julia_history
+```
+
+### `JULIA_PKGRESOLVE_ACCURACY`
+
+A positive `Int` that determines how much time the max-sum subroutine
+`MaxSum.maxsum()` of the package dependency resolver
+[`Base.Pkg.resolve`](@ref) will devote to attempting
+satisfying constraints before giving up: this value is by default `1`, and
+larger values correspond to larger amounts of time.
+
+Suppose the value of `$JULIA_PKGRESOLVE_ACCURACY` is `n`. Then
+
+*   the number of pre-decimation iterations is `20*n`,
+*   the number of iterations between decimation steps is `10*n`, and
+*   at decimation steps, at most one in every `20*n` packages is decimated.
+
+## External applications
+
+### `JULIA_SHELL`
+
+The absolute path of the shell with which Julia should execute external
+commands (via `Base.repl_cmd()`). Defaults to the environment variable
+`$SHELL`, and falls back to `/bin/sh` if `$SHELL` is unset.
+
+!!! note the `fish` shell is unsupported.
+
+### `JULIA_EDITOR`, `VISUAL`, `EDITOR`
```

Comment:
I would take `VISUAL` and `EDITOR` out of the heading as they aren't Julia specific ENV variables.